src/lib.rs

```diff
@@ -25,7 +25,7 @@ pub enum LoadingError {
     RustdocIoError(String, std::io::Error),
 
     #[error("unable to detect rustdoc 'format_version' key in file: {0}")]
-    RustdocFormatDetection(String, anyhow::Error),
```

Comment:
I think we still want an error saying what specifically is wrong with the file. Otherwise we won't know which of the many `RustdocFormatDetection` error lines triggered, and debugging will be quite frustrating.

You can construct an `anyhow::Error` with `anyhow::anyhow!()`.